python/lsst/ts/electrometer/commands_factory.py

```diff
@@ -407,7 +493,10 @@ def set_mode(self, mode):
         command : `str`
             The generated command string.
         """
-        command = f":sens:func '{enums.UnitMode(mode)}';"
+        if self.brand == "Keysight":
+            command = f":sens:func '{enums.UnitMode(mode).name}';"
+        else:
+            command = f":sens:func '{enums.UnitMode(mode)}';"

```

Comment:
I think perhaps the Keithley will work with the `enums.UnitMode(mode).name` which would get rid of the need for an `if`/`else` statement here. We should check